<!-- Launch the generator node -->
<launch>
    <node name="generator1" pkg="simulator" type="generator.py" output="screen">
        <param name="node_id" value="11"/>
        <param name="node_name" value="generator1"/>
        <param name="next_element" value="switch"/>
        <param name="gen_freq" value="1.0"/>
        <param name="num_messages" value="10"/>
        <param name="event_type" value="corriere"/>
        <param name="attribute1" value="standard"/>
        <param name="value1" value="0.3"/>
        
    </node>

    <node name="generator2" pkg="simulator" type="generator.py" output="screen">
        <param name="node_id" value="12"/>
        <param name="node_name" value="generator2"/>
        <param name="next_element" value="switch"/>
        <param name="gen_freq" value="1.0"/>
        <param name="num_messages" value="20"/>
        <param name="event_type" value="multicollo"/>
        <param name="attribute1" value="standard"/>
        <param name="value1" value="0.2"/>
    </node>

    <node name="switch" pkg="simulator" type="switch2.py" output="screen">
        <param name="node_id" value="101"/>
        <param name="node_name" value="switch"/>
        <param name="modality" value="attribute1_split"/>
        <param name="topic1_attribute" value="standard"/>
        <param name="topic1" value="standard"/>
        <param name="topic2" value="geek"/>
    </node>
        
    <node name="geek" pkg="simulator" type="blocco_base.py" output="screen">
        <param name="node_id" value="21"/>
        <param name="node_name" value="geek"/>
        <param name="next_element" value="switch2"/>
        <param name="num_servers" value="2"/>
        <param name="server_time" value="0.3"/>
    </node>

    <node name="standard" pkg="simulator" type="blocco_base.py" output="screen">
        <param name="node_id" value="21"/>
        <param name="node_name" value="standard"/>
        <param name="next_element" value="switch2"/>
        <param name="num_servers" value="2"/>
        <param name="server_time" value="5"/>
    </node>

    <node name="switch2" pkg="simulator" type="switch2.py" output="screen">
        <param name="node_id" value="101"/>
        <param name="node_name" value="switch2"/>
        <param name="modality" value="type"/>
        <param name="topic1" value="multicollo"/>
        <param name="topic2" value="corriere"/>
        <param name="topic1_attribute" value="multicollo"/>
    
    </node>

    <node name="multicollo" pkg="simulator" type="blocco_base.py" output="screen">
        <param name="node_id" value="21"/>
        <param name="node_name" value="multicollo"/>
        <param name="next_element" value="end_node"/>
        <param name="num_servers" value="2"/>
        <param name="server_time" value="2"/>
    </node>

    <node name="corriere" pkg="simulator" type="blocco_base.py" output="screen">
        <param name="node_id" value="22"/>
        <param name="node_name" value="corriere"/>
        <param name="next_element" value="end_node"/>
        <param name="num_servers" value="3"/>
        <param name="server_time" value="4"/>
    </node>

    <node name="end_node" pkg="simulator" type="end_node.py" output="screen">
        <param name="node_id" value="7"/>
        <param name="node_name" value="end_node"/>
    </node>

</launch>